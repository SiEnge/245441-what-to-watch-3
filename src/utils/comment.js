export const parseDateComment = (date) => {
  return new Date(date).toLocaleString(`en-us`, {
    month: `long`,
    year: `numeric`,
    day: `numeric`,
  });
};
