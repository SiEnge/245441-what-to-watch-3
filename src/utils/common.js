import {MINUTE_IN_ONE_HOUR, TEN_MINUTES, ScoreType} from "../const.js";

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const getScoreType = (score) => {
  let scoreName;

  if (score >= 0 && score < 3) {
    scoreName = ScoreType.BAD;
  } else if (score >= 3 && score < 5) {
    scoreName = ScoreType.NORMAL;
  } else if (score >= 5 && score < 8) {
    scoreName = ScoreType.GOOD;
  } else if (score >= 8 && score < 10) {
    scoreName = ScoreType.VERY_GOOD;
  } else if (score >= 10) {
    scoreName = ScoreType.AWESOME;
  }

  return scoreName;
};

export const parseDuration = (duration) => {
  if (duration > MINUTE_IN_ONE_HOUR) {
    const hour = Math.floor(duration / MINUTE_IN_ONE_HOUR);
    const minute = (duration % MINUTE_IN_ONE_HOUR < TEN_MINUTES) ? `0${duration % MINUTE_IN_ONE_HOUR}` : duration % MINUTE_IN_ONE_HOUR;
    return `${hour}h ${minute}m`;
  }

  return `${duration}m`;
};

export const parseTime = (duration) => {
  let hour = `00`;
  let minut = `00`;
  let second = `00`;

  if (+duration < 60 && +duration !== 0) {
    second = duration;
  } else {
    minut = String(Math.floor(+duration / 60));
    second = String(+duration - (+minut * 60));
  }

  if (+minut > 60) {
    hour = String(Math.floor(+minut / 60));
    minut = String(+minut - (+hour * 60));
  }

  if (hour.length === 1) {
    hour = `0${hour}`;
  }

  if (minut.length === 1) {
    minut = `0${minut}`;
  }

  if (second.length === 1) {
    second = `0${second}`;
  }

  return `${hour}:${minut}:${second}`;
};
