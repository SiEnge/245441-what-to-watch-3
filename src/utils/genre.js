const MAX_COUNT_GENRES = 9;

export const getGenres = (movies) => {
  const genres = new Set();

  movies.forEach((movie) => genres.add(movie.genre));
  return Array.from(genres).slice(0, MAX_COUNT_GENRES);
};

