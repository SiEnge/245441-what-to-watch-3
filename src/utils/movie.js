export const adapterMovie = (movie) => {
  return {
    id: movie.id,
    title: movie.name,
    poster: movie.poster_image,
    previewImage: movie.preview_image,
    videoLink: movie.video_link,
    previewVideo: movie.preview_video_link,
    genre: movie.genre,
    date: String(movie.released),
    background: movie.background_image,
    backgroundColor: movie.background_color, // добавить в пропс и вывод в UI
    descriptions: movie.description.split(". "),
    score: movie.scores_count,
    rating: movie.rating,
    runtime: movie.run_time,
    director: movie.director,
    starring: movie.starring,
    isFavorite: movie.is_favorite,
  };
};

export const adapterMovies = (movies) => {
  return movies.map((movie) => adapterMovie(movie));
};

