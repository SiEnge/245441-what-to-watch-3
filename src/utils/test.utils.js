export const promoMovieServer = {
  'id': 1,
  'name': `The Grand Budapest Hotel`,
  'poster_image': `img/the-grand-budapest-hotel-poster.jpg`,
  'preview_image': `img/the-grand-budapest-hotel.jpg`,
  'background_image': `img/the-grand-budapest-hotel-bg.jpg`,
  'background_color': `#ffffff`,
  'video_link': `https://some-link`,
  'preview_video_link': `https://some-link`,
  'description': `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  'rating': 8.9,
  'scores_count': 240,
  'director': `Wes Andreson`,
  'starring': [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`],
  'run_time': 99,
  'genre': `Comedy`,
  'released': 2014,
  'is_favorite': false
};

export const movies = [
  {
    id: 1,
    title: `The Grand Budapest Hotel`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    videoLink: `https://some-link`,
    previewVideo: `https://some-link`,
    previewImage: `img/the-grand-budapest-hotel.jpg`,
    genre: `Comedy`,
    date: `2014`,
    background: `img/the-grand-budapest-hotel-bg.jpg`,
    backgroundColor: `#ffffff`,
    descriptions: [
      `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H`,
      `Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    ],
    rating: 8.9,
    score: 240,
    runtime: 99,
    isFavorite: false,
    director: `Wes Andreson`,
    starring: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`,
    ],
  },
  {
    id: 2,
    title: `film2`,
    poster: `img/film2.jpg`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideo: `img/the-grand-budapest-hotel.jpg`,
    previewImage: `img/the-grand-budapest-hotel.jpg`,
    genre: `Comedy`,
    date: `2014`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    descriptions: [
      `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
      `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    ],
    rating: 6.6,
    score: 240,
    isFavorite: false,
    runtime: 126,
    director: `Wes Andreson`,
    starring: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`,
      `Tony Revoloru`,
      `Tilda Swinton`,
      `Tom Wilkinson`,
      `Owen Wilkinson`,
      `Adrien Brody`,
      `Ralph Fiennes`,
      `Jeff Goldblum`
    ],
  },
  {
    id: 3,
    title: `film3`,
    poster: `img/film3.jpg`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideo: `img/the-grand-budapest-hotel.jpg`,
    previewImage: `img/the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    date: `2014`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    descriptions: [
      `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
      `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    ],
    rating: 7.5,
    score: 240,
    isFavorite: false,
    runtime: 126,
    director: `Wes Andreson`,
    starring: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`,
      `Tony Revoloru`,
      `Tilda Swinton`,
      `Tom Wilkinson`,
      `Owen Wilkinson`,
      `Adrien Brody`,
      `Ralph Fiennes`,
      `Jeff Goldblum`
    ],
  },
  {
    id: 4,
    title: `film4`,
    poster: `img/film4.jpg`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideo: `img/the-grand-budapest-hotel.jpg`,
    previewImage: `img/the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    date: `2014`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    descriptions: [
      `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
      `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    ],
    rating: 16.6,
    score: 240,
    isFavorite: false,
    runtime: 126,
    director: `Wes Andreson`,
    starring: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`,
      `Tony Revoloru`,
      `Tilda Swinton`,
      `Tom Wilkinson`,
      `Owen Wilkinson`,
      `Adrien Brody`,
      `Ralph Fiennes`,
      `Jeff Goldblum`
    ],
  },
  {
    id: 5,
    title: `film5`,
    poster: `img/film5.jpg`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideo: `img/the-grand-budapest-hotel.jpg`,
    previewImage: `img/the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    date: `2014`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    descriptions: [
      `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
      `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    ],
    rating: 4.6,
    score: 240,
    isFavorite: false,
    runtime: 126,
    director: `Wes Andreson`,
    starring: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`,
      `Tony Revoloru`,
      `Tilda Swinton`,
      `Tom Wilkinson`,
      `Owen Wilkinson`,
      `Adrien Brody`,
      `Ralph Fiennes`,
      `Jeff Goldblum`
    ],
  },
  {
    id: 6,
    title: `film6`,
    poster: `img/film6.jpg`,
    videoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideo: `img/the-grand-budapest-hotel.jpg`,
    previewImage: `img/the-grand-budapest-hotel.jpg`,
    genre: `Drama`,
    date: `2014`,
    background: `img/bg-the-grand-budapest-hotel.jpg`,
    backgroundColor: `#ffffff`,
    descriptions: [
      `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
      `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    ],
    rating: 5.3,
    score: 240,
    isFavorite: false,
    runtime: 126,
    director: `Wes Andreson`,
    starring: [
      `Bill Murray`,
      `Edward Norton`,
      `Jude Law`,
      `Willem Dafoe`,
      `Saoirse Ronan`,
      `Tony Revoloru`,
      `Tilda Swinton`,
      `Tom Wilkinson`,
      `Owen Wilkinson`,
      `Adrien Brody`,
      `Ralph Fiennes`,
      `Jeff Goldblum`
    ],
  }
];

export const movie = movies[0];

export const comments = [
  {
    id: 1,
    userName: `Kate Muir`,
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    userName: `Kate Muir`,
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 3,
    userName: `Kate Muir`,
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
];

export const comment = comments[0];

export const classLinkLogo = `logo__link`;

export const user = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`,
};

export const serverUser = {
  'id': 1,
  'email': `Oliver.conner@gmail.com`,
  'name': `Oliver.conner`,
  'avatar_url': `img/1.png`,
};

export const genres = [
  `all`,
  `Comedies`,
  `Crime`,
  `Documentary`,
  `Dramas`,
  `Horror`,
  `Romance`,
  `Sci-Fi`,
  `Thrillers`,
];

export const activeGenre = `Horror`;

export const activeTab = `Overview`;

