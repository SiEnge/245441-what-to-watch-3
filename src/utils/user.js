export const adapterUser = (user) => {
  return {
    id: user.id,
    email: user.email,
    name: user.name,
    avatarUrl: user.avatar_url,
  };
};
