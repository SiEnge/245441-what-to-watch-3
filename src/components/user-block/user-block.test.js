import React from "react";
import renderer from "react-test-renderer";
import {Router} from "react-router-dom";
import UserBlock from "./user-block.jsx";
import history from "../../history.js";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space";
import {user} from "../../utils/test.utils.js";

const mockStore = configureStore([]);

const store = mockStore({
  [NameSpace.USER]: {
    user,
  }
});

it(`Render UserBlock with authorization`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <UserBlock
              isAuth={true}
              user={user}
            />
          </Router>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it(`Render UserBlock without authorization`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router history={history}>
            <UserBlock
              isAuth={false}
              user={user}
            />
          </Router>
        </Provider>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
