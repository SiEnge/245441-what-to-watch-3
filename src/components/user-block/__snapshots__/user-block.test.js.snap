// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render UserBlock with authorization 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <a
      href="/mylist"
      onClick={[Function]}
    >
      <img
        alt="User avatar"
        height="63"
        src="https://htmlacademy-react-3.appspot.com/img/1.png"
        width="63"
      />
    </a>
  </div>
</div>
`;

exports[`Render UserBlock without authorization 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
